src/main/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoff.java

```diff
@@ -0,0 +1,211 @@
+            new ArrayList(Arrays.asList(500, 502, 503, 504));
+
+    private SphereClientConfig clientConfig;
+    private long maxDelay;
+    private long initialRetryDelay;
+    private int maxRetryAttempt;
+    private int maxParallelRequests;
+    private List<Integer> statusCodesToRetry;
+
+    private RetryableSphereClientWithExponentialBackoff(@Nonnull final SphereClientConfig clientConfig) {
+        this.clientConfig = clientConfig;
+        this.maxDelay = DEFAULT_MAX_DELAY;
+        this.initialRetryDelay = DEFAULT_INITIAL_RETRY_DELAY;
+        this.maxRetryAttempt = DEFAULT_MAX_RETRY_ATTEMPT;
+        this.maxParallelRequests = DEFAULT_MAX_PARALLEL_REQUESTS;
+        this.statusCodesToRetry = DEFAULT_STATUS_CODES_TO_RETRY;
+    }
+
+    /**
+     * Creates a new instance of {@link RetryableSphereClientWithExponentialBackoff} given a {@link SphereClientConfig}
+     * responsible for creation of a SphereClient.
+     *
+     * @param clientConfig the client configuration for the client.
+     * @return the instantiated {@link RetryableSphereClientWithExponentialBackoff}.
+     */
+    public static RetryableSphereClientWithExponentialBackoff of(
+            @Nonnull final SphereClientConfig clientConfig) {
+        return new RetryableSphereClientWithExponentialBackoff(clientConfig);
+    }
+
+    /**
+     * Sets the maxDelay value value in milliseconds.
+     * @param maxDelay - build with maxDelay value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given maxDelay value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withMaxDelay(final long maxDelay) {
+        this.maxDelay = maxDelay;
+        return this;
+    }
+
+    /**
+     * Sets the initialDelay value in milliseconds.
+     * @param initialDelay - build with initialDelay value.
+     * If initialDelay is less than maxDelay then, a {@link IllegalArgumentException} will be thrown.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given initialDelay value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withInitialDelay(final long initialDelay) {
+        if (initialDelay < maxDelay) {
+            this.initialRetryDelay = initialDelay;
+        } else {
+            throw new IllegalArgumentException(
+                    format("InitialDelay %s is less than MaxDelay %s.", initialDelay, maxDelay));
+        }
+        return this;
+    }
+
+    /**
+     * Sets the Max Retry value, It should be greater than 1 for the Retry attempt.
+     * @param maxRetryAttempt - build with maxRetries value.
+     * If maxRetryAttempt is less than 1 then, a {@link IllegalArgumentException} will be thrown.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given maxRetries value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withMaxRetryAttempt(final int maxRetryAttempt) {
+        if (maxRetryAttempt > 1) {
```

Comment:
```suggestion
        if (maxRetryAttempt > 0) {
```

1 retry is ok because it's 1 retry, but 0 or less than 0 does not make sense. You have a test case for 1 retry :)

Reply:
Done. https://github.com/commercetools/commercetools-sync-java/pull/623/commits/9eeb739ad4c18207082c9fefe27bd8aca4f5df00